### ISSUE TITLE
K-L metric or loss

### ISSUE THREAD

**Issue by dbuscombe-usgs** [dbuscombe-usgs/segmentation_zoo#35]
it makes sense to explore K-L distance as a loss function or metric, see:

https://github.com/dbuscombe-usgs/segmentation_zoo/issues/32#issuecomment-966740690

**Comment by dbuscombe-usgs** [comment]
`plotcomp_n_getiou` now computes and reports K-L divergence for each validation and train pair tested in `train_model.py`

![hatteras_l8_resunet_val_15](https://user-images.githubusercontent.com/3596509/141397620-8dc6d4a9-6e53-47c6-96d9-fee5772fb096.png)

![hatteras_l8_resunet_val_52](https://user-images.githubusercontent.com/3596509/141397656-fc173dc1-7d3b-4f2f-810d-746c369097d7.png)

smaller values of KLD are better (smaller divergence)

![D_vs_KL](https://user-images.githubusercontent.com/3596509/141398244-0c2b15dd-1bf2-4913-ae0c-d8b0d047238d.png)


Hatteras stats now:
```
Evaluating model on entire validation set ...
15/15 [==============================] - 2s 164ms/step - loss: 0.0222 - mean_iou: 0.9839 - dice_coef: 0.9778
loss=0.0222, Mean IOU=0.9839, Mean Dice=0.9778
Mean of mean IoUs (validation subset)=0.983
Mean of mean Dice scores (validation subset)=0.977
Mean of mean KLD scores (train subset)=0.142
Mean of mean IoUs (train subset)=0.988
Mean of mean Dice scores (train subset)=0.982
Mean of mean KLD scores (train subset)=0.100
```

also added KLD as a model training option

```
elif LOSS.startswith('k'):
    model.compile(optimizer = 'adam', loss =tf.keras.losses.KLDivergence(), metrics = [mean_iou, dice_coef])
```

**Comment by dbuscombe-usgs** [comment]
needs:

- [ ] test model training with KL-divergence
- [ ] docs

**Comment by dbuscombe-usgs** [comment]
trained resunet model with Kullback–Leibler on the Klamath L8 C/T imagery with 80% validation split and 0.2 dropout

As I suspected, the unet does converge easily and quickly

![ct_E_resunet_allclasses_512_remap_trainhist_8](https://user-images.githubusercontent.com/3596509/141411379-17aef824-19e6-4044-af8d-22a26fc9ac4d.png)

outputs are good, comparable to dice loss

perhaps  it is a good loss function for a Unet when you want to preserve the overall frequency distribution of classes in model predictions?

**Comment by dbuscombe-usgs** [comment]
tested, docs updated. closing
